transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -315,7 +315,7 @@ static jstring netty_epoll_native_kernelVersion(JNIEnv* env, jclass clazz) {
 }
 
 static jboolean netty_epoll_native_isSupportingSendmmsg(JNIEnv* env, jclass clazz) {
-    if (sendmmsg) {
+    if (&sendmmsg) {
```

Comment:
But this solution doesn't work.  The symbol is *not* weakly defined.  If `IO_NETTY_SENDMMSG_NOT_FOUND` is set, it's a compile time error for redefining it:

```
#define _GNU_SOURCE
#include <sys/socket.h>

struct mmsghdr {
    struct msghdr msg_hdr;  /* Message header */
    unsigned int  msg_len;  /* Number of bytes transmitted */
};

extern int sendmmsg(int sockfd, struct mmsghdr* msgvec, unsigned int vlen, unsigned int flags) __attribute__((weak));

int main(int argc, char ** argv) {
  return 0;
}
```
```
Wed Jun 21 03:02:12$ gcc send.c 
send.c:5:8: error: redefinition of ‘struct mmsghdr’
 struct mmsghdr {
        ^
In file included from send.c:2:0:
/usr/include/x86_64-linux-gnu/sys/socket.h:101:8: note: originally defined here
 struct mmsghdr
        ^
send.c:12:12: error: conflicting types for ‘sendmmsg’
 extern int sendmmsg(int sockfd, struct mmsghdr* msgvec, unsigned int vlen, unsigned int flags) __attribute__((weak));
            ^
In file included from send.c:2:0:
/usr/include/x86_64-linux-gnu/sys/socket.h:193:12: note: previous declaration of ‘sendmmsg’ was here
 extern int sendmmsg (int __fd, struct mmsghdr *__vmessages,
            ^
```

Also, to be clear, sendmmsg is *NOT* a function pointer, it is a function.  It cannot be null if it is defined.  Did you try compiling with `-Wpointer-bool-conversion` ? 

The real issue is that this file is selectively defining the method as weak, which is the true problem

I talked with several other C++ people before I sent this out, and this seems to be the least worst solution.  The next bets solution would be something like:


```
int (*sendmmsg_fp)(int sockfd, struct mmsghdr* msgvec, unsigned int vlen, unsigned int flags);

#ifdef IO_NETTY_SENDMMSG_NOT_FOUND
extern int sendmmsg(int sockfd, struct mmsghdr* msgvec, unsigned int vlen, unsigned int flags) __attribute__((weak));
sendmmsg_fp = sendmmsg;

#ifndef __USE_GNU
struct mmsghdr {
    struct msghdr msg_hdr;  /* Message header */
    unsigned int  msg_len;  /* Number of bytes transmitted */
};
#endif
#else
sendmmsg_fp = sendmmsg;
#endif
```

Then, everywhere else in this file, use `sendmmsg_fp`.  However, that's a lot more work, and not as clean.